src/commcare_cloud/commands/inventory_lookup/inventory_lookup.py

```diff
@@ -118,24 +126,21 @@ def run(self, args, ssh_args):
             # Always include ssh agent forwarding on control machine
             ssh_args = ['-A'] + ssh_args
 
-            if os.environ.get('COMMCARE_CLOUD_USE_AWS_SSM') != '0' and \
-                    is_aws_env(environment) and \
-                    not is_ec2_instance_in_account(environment.aws_config.sso_config.sso_account_id):
-                if not is_session_manager_plugin_installed():
-                    puts(color_error(
-                        "Before you can use AWS SSM to connect, you "
-                        "must install the AWS session-manager-plugin"
-                    ))
-                    puts(
-                        f"{color_notice('See ')}{color_link(self.INSTALL_SSM)}"
-                        f"{color_notice(' for instructions.')}"
-                    )
-                    return -1
-                use_aws_ssm_with_instance_id = environment.get_host_vars(
-                    environment.groups['control'][0]
-                )['ec2_instance_id']
-                env_vars = os.environ.copy()
-                env_vars.update({'AWS_PROFILE': aws_sign_in(environment)})
+        if self.should_use_ssm(environment):
+            if not is_session_manager_plugin_installed():
+                puts(color_error(
+                    "Before you can use AWS SSM to connect, you "
+                    "must install the AWS session-manager-plugin"
+                ))
+                puts(
+                    f"{color_notice('See ')}{color_link(self.INSTALL_SSM)}"
+                    f"{color_notice(' for instructions.')}"
+                )
+                return -1
+            _, host, _ = self.get_address_tuple(args)
+            use_aws_ssm_with_instance_id = environment.get_host_vars(host)['ec2_instance_id']
+            env_vars = os.environ.copy()
+            env_vars.update({'AWS_PROFILE': aws_sign_in(environment)})
```

Comment:
Is there a reason we can't use the aws `--profile` option instead? (I've been meaning to look into this for a while now.)

```bash
$ aws ssm -h
usage: aws [-h] [--profile PROFILE] [--debug]
...
```

Reply:
I'm not sure. This code did not change other than the if-block was un-indented.

Reply:
I should have pointed out earlier that the commit where this "changed" is easier to review with whitespace changes hidden:

https://github.com/dimagi/commcare-cloud/pull/5271/commits/e9ba2770aa2765d58cf5183c5f466934a249e382?diff=unified&w=1